modules/core/src/main/scala/com.snowplowanalytics.iglu/client/IgluCirceClient.scala

```diff
@@ -43,13 +44,17 @@ final class IgluCirceClient[F[_]] private (
     M: Monad[F],
     L: RegistryLookup[F],
     C: Clock[F]
-  ): EitherT[F, ClientError, Unit] =
+  ): EitherT[F, (ClientError, SupersededBy), SupersededBy] =
```

Comment:
Instead of returning a tuple here, did you consider amending `ClientError` to carry the extra information?  Can you share an example of the failed event that will get created by Enrich if `SupersededBy` is non-empty?

Reply:
`SupersededBy` might be non-empty when there is a validation error. For that case, we might add `supersededBy` field on `ClientError.ValidationError` instead of returning it in a tuple.